accounts/tests/test_user.py

```diff
@@ -178,73 +180,209 @@ def test_user_delete_make_invalid_password(self):
-        resp = self.client.post(reverse('accounts-delete'),
-                                {'encrypted_link': encr_link, 'password': 'testpass', 'delete_sounds': 'delete_sounds'})
+        self.client.post(reverse('accounts-delete'),
+                         {'encrypted_link': encr_link, 'password': 'testpass', 'delete_sounds': 'delete_sounds'})
 
-        self.assertEqual(User.objects.get(id=user.id).profile.is_deleted_user, True)
+        # Check user is marked as anonymized
+        self.assertTrue(User.objects.get(id=user.id).profile.is_anonymized_user)
+
+        # Check sounds have been deleted as well
+        self.assertFalse(Sound.objects.filter(user__id=user.id).exists())
+
+        # Check DeletedUser object has been created for this user with the "self deleted" reason
+        self.assertEqual(DeletedUser.objects.get(user_id=user.id).reason, DeletedUser.DELETION_REASON_SELF_DELETED)
+
+    def test_user_delete_keep_sounds_using_web_form(self):
+        # Test user's option to delete user account but preserving the sounds
 
-    def test_fail_user_delete_using_form(self):
-        # This should try to delete the account but with a wrong password
         user = self.create_user_and_content(is_index_dirty=False)
-        a = self.client.login(username=user.username, password='testpass')
+        self.client.login(username=user.username, password='testpass')
         form = DeleteUserForm(user_id=user.id)
         encr_link = form.initial['encrypted_link']
-        resp = self.client.post(reverse('accounts-delete'),
-                                {'encrypted_link': encr_link, 'password': 'wrong_pass',
-                                 'delete_sounds': 'delete_sounds'})
+        self.client.post(reverse('accounts-delete'),
+                         {'encrypted_link': encr_link, 'password': 'testpass', 'delete_sounds': 'only_user'})
+
+        # Check user is marked as anonymized
+        self.assertTrue(User.objects.get(id=user.id).profile.is_anonymized_user)
+
+        # Check sounds have not been deleted
+        self.assertTrue(Sound.objects.filter(user__id=user.id).exists())
 
-        self.assertEqual(User.objects.get(id=user.id).profile.is_deleted_user, False)
+        # Check DeletedUser object has been created for this user with the "self deleted" reason
+        self.assertEqual(DeletedUser.objects.get(user_id=user.id).reason, DeletedUser.DELETION_REASON_SELF_DELETED)
+
+    def test_fail_user_delete_include_sounds_using_web_form(self):
+        # Test delete user account form with wrong password does not delete
+
+        # This should try to delete the account but with a wrong password
+        user = self.create_user_and_content(is_index_dirty=False)
+        self.client.login(username=user.username, password='testpass')
+        form = DeleteUserForm(user_id=user.id)
+        encr_link = form.initial['encrypted_link']
+        self.client.post(reverse('accounts-delete'),
```

Comment:
we could test that the returned text doesn't have "you've deleted your stuff", or that it includes "your password is wrong", or whatever other message you put here

Reply:
Addressed e769296c26d70aba8fadfd232402727c7dfc014b